python-frontend/src/main/java/org/sonar/python/semantic/v2/typeshed/VarSymbolToDescriptorConverter.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.semantic.v2.typeshed;
+
+import java.util.Optional;
+import org.sonar.python.index.Descriptor;
+import org.sonar.python.index.VariableDescriptor;
+import org.sonar.python.types.protobuf.SymbolsProtos;
+
+public class VarSymbolToDescriptorConverter {
+
+  public Descriptor convert(SymbolsProtos.VarSymbol varSymbol) {
+    var typeAnnotation = Optional.of(varSymbol)
+      .map(SymbolsProtos.VarSymbol::getTypeAnnotation)
+      .map(SymbolsProtos.Type::getFullyQualifiedName)
+      .orElse(null);
+    return new VariableDescriptor(varSymbol.getName(), varSymbol.getFullyQualifiedName(), typeAnnotation);
```

Comment:
When looking at what is done in `SymbolImpl`, there is a step of normalization of FQNs that removes the `builtins` prefix.

I'm personally in favor of keeping this prefix, however I have the feeling the change of behavior here is not intentional (and if we decide to add back the `builtins` prefix we need to do it everywhere anyway).

On top of that, there is some logic about FQNs when they are from a class, which I'm not sure whether we need to take care here or not (it depends on what is returned by the protobuf for some sample variables - which would benefit from being tested).